File: backend/GameApp.Application/Services/ItemServices/ItemGetService.cs
using GameApp.Application.UseCases.ItemUseCases;
using GameApp.Domain.Entities.Items;
using GameApp.Domain.Repositories;
using GameApp.Domain.ValueObjects.Items;

namespace GameApp.Application.Services.ItemServices;


public class ItemGetService : ItemGetUseCase
{
    private readonly IItemRepository _repo;

    public ItemGetService(IItemRepository repo) => _repo = repo;

    public async Task<Item?> GetItem(Guid id)
    {
        return await _repo.FetchByIdAsync(id);
    }

    public async Task<Item?> GetItemByName(ItemName name)
    {
        return await _repo.FetchByName(name);
    }

    public async Task<IEnumerable<Item>> GetAllItems()
    {
        return await _repo.FetchAllAsync();
    }
}

File: frontend/gameapp-frontend/src/application/services/item-services/item-get-service.js
import { IItemGetUseCase } from "../../usecases/item-use-cases/item-get-use-case";
import { IitemRepository } from "../../repositories/iitem-repository";

export class ItemGetService extends IItemGetUseCase {
    constructor(itemRepository) {
        super();
        if (!(itemRepository instanceof IitemRepository)) {
            throw new TypeError("itemRepository must be an instance of IitemRepository");
        }
        this.itemRepository = itemRepository;
    }

    async getItem(id) {
        if (!id) throw new TypeError("id is required");
        return await this.itemRepository.fetchById(id);
    }

    async getItemByName(name) {
        if (!name) throw new TypeError("name is required");
        return await this.itemRepository.fetchByName(name);
    }

    async getAllItems() {
        return await this.itemRepository.fetchAll();
    }
}
